sql/src/test/java/org/apache/druid/sql/calcite/BaseCalciteQueryTest.java

```diff
@@ -138,6 +138,12 @@ public static void setupNullValues()
     NULL_STRING = NullHandling.defaultStringValue();
     NULL_FLOAT = NullHandling.defaultFloatValue();
     NULL_LONG = NullHandling.defaultLongValue();
+    NULL_LIST = new ArrayList<Object>()
+    {
+      {
+        add(null);
+      }
+    };
```

Comment:
hmm, I know this is because of the coercion behavior in `ExprEval`, but I'm having some regret about the decision to homogenize `null`, `[]`, and `[null]` to `[null]`. I think it made the most sense when mapping back to `STRING` type, but I think the coercion logic should probably be different when we are grouping on a multi-value string as an ARRAY<STRING> instead of a STRING, and the input bindings should homogenize the value to `null` instead in that case.

We don't need to change this right now I think, since this PR still isn't documenting this functionality, but something to think about for the future I think, since `[null]` is sort of strange on the result side of things when stuff is left as an actual array